migrator/migrations/m_192_to_m_193_create_vulnerability_requests_for_legacy/migration_impl.go

```diff
@@ -0,0 +1,298 @@
+package m192tom193
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	timestamp "github.com/gogo/protobuf/types"
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/migrator/migrations/m_192_to_m_193_create_vulnerability_requests_for_legacy/schema"
+	"github.com/stackrox/rox/migrator/types"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/postgres/pgutils"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/uuid"
+	"gorm.io/gorm/clause"
+)
+
+var (
+	batchSize = 2000
+	log       = logging.LoggerForModule()
+
+	sysUser = &storage.SlimUser{
+		Id:   uuid.NewV4().String(),
+		Name: "system",
+	}
+)
+
+func migrate(database *types.Databases) error {
+	ctx := sac.WithAllAccess(context.Background())
+	pgutils.CreateTableFromModel(ctx, database.GormDB, schema.CreateTableImagesStmt)
+	pgutils.CreateTableFromModel(ctx, database.GormDB, schema.CreateTableImageCvesStmt)
+	pgutils.CreateTableFromModel(ctx, database.GormDB, schema.CreateTableImageCveEdgesStmt)
+	pgutils.CreateTableFromModel(ctx, database.GormDB, schema.CreateTableVulnerabilityRequestsStmt)
+
+	now := timestamp.TimestampNow()
+	snoozedCVEMap, err := collectSnoozedImageCVEs(ctx, database)
+	if err != nil || len(snoozedCVEMap) == 0 {
+		return err
+	}
+	if err = createVulnRequests(ctx, database, now, snoozedCVEMap); err != nil {
+		return err
+	}
+	if err = updateImageCVEEdges(ctx, database, snoozedCVEMap); err != nil {
+		return err
+	}
+	return updateImageCVEs(ctx, database)
+}
+
+func collectSnoozedImageCVEs(ctx context.Context, database *types.Databases) (map[string]*timestamp.Timestamp, error) {
+	query := database.GormDB.WithContext(ctx).Table(schema.ImageCvesTableName).
```

Comment:
Nice.  Thats one advantage of using gorm is that it is easier to use certain filtering techniques.